File: fm-domain/src/main/java/io/zeitwert/fm/collaboration/model/enums/CodeNoteType.java
package io.zeitwert.fm.collaboration.model.enums;

import io.dddrive.core.enums.model.Enumeration;
import io.dddrive.core.enums.model.base.EnumeratedBase;

/**
 * CodeNoteType enum using the NEW dddrive framework.
 */
public final class CodeNoteType extends EnumeratedBase {

    public CodeNoteType(Enumeration<CodeNoteType> enumeration, String id, String name) {
        super(enumeration, id, name);
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (other == null || getClass() != other.getClass()) return false;
        CodeNoteType that = (CodeNoteType) other;
        return getId().equals(that.getId());
    }

    @Override
    public int hashCode() {
        return getId().hashCode();
    }
}

File: fm-domain/src/main/java/io/zeitwert/fm/collaboration/model/enums/CodeNoteType.kt
package io.zeitwert.fm.collaboration.model.enums

import dddrive.ddd.model.EnumeratedEnum
import dddrive.ddd.model.base.EnumerationBase

/** Note type enum using the NEW dddrive framework. */
enum class CodeNoteType(
	override val defaultName: String,
) : EnumeratedEnum {

	NOTE("Note"),
	CALL("Call"),
	VISIT("Visit"),
	;

	override val enumeration get() = Enumeration

	companion object Enumeration : EnumerationBase<CodeNoteType>(CodeNoteType::class.java) {
		init {
			entries.forEach { addItem(it) }
		}

		@JvmStatic
		fun getNoteType(itemId: String?) = if (itemId != null) getItem(itemId) else null
	}
}
